src/main/java/net/java/otr4j/session/SessionKeys.java

```diff
@@ -1,54 +1,230 @@
+
 package net.java.otr4j.session;
 
 import java.math.BigInteger;
+import java.nio.ByteBuffer;
 import java.security.KeyPair;
+import java.util.Arrays;
+import java.util.logging.Logger;
 
 import javax.crypto.interfaces.DHPublicKey;
 
 import net.java.otr4j.OtrException;
-
-public interface SessionKeys {
-
-	public static final int Previous = 0;
-	public static final int Current = 1;
-	public static final byte HIGH_SEND_BYTE = (byte) 0x01;
-	public static final byte HIGH_RECEIVE_BYTE = (byte) 0x02;
-	public static final byte LOW_SEND_BYTE = (byte) 0x02;
-	public static final byte LOW_RECEIVE_BYTE = (byte) 0x01;
-
-	public abstract void setLocalPair(KeyPair keyPair, int localPairKeyID);
-
-	public abstract void setRemoteDHPublicKey(DHPublicKey pubKey,
-			int remoteKeyID);
-
-	public abstract void incrementSendingCtr();
-
-	public abstract byte[] getSendingCtr();
-
-	public abstract byte[] getReceivingCtr();
-
-	public abstract void setReceivingCtr(byte[] ctr);
-
-	public abstract byte[] getSendingAESKey() throws OtrException;
-
-	public abstract byte[] getReceivingAESKey() throws OtrException;
-
-	public abstract byte[] getSendingMACKey() throws OtrException;
-
-	public abstract byte[] getReceivingMACKey() throws OtrException;
-
-	public abstract void setS(BigInteger s);
-
-	public abstract void setIsUsedReceivingMACKey(Boolean isUsedReceivingMACKey);
-
-	public abstract Boolean getIsUsedReceivingMACKey();
-
-	public abstract int getLocalKeyID();
-
-	public abstract int getRemoteKeyID();
-
-	public abstract DHPublicKey getRemoteKey();
-
-	public abstract KeyPair getLocalPair();
-
-}
\ No newline at end of file
+import net.java.otr4j.crypto.OtrCryptoEngine;
+import net.java.otr4j.io.SerializationUtils;
+
+public class SessionKeys {

```

Comment:
Please migrate/add original "@author" comment of this class (it's in SessionKeysImpl)